packages/components/demo/example/example.component.ts

```diff
@@ -0,0 +1,46 @@
+import { Component, OnInit } from "@angular/core";
+import { TemplateBaseComponent } from "../../../../src/app/shared/components/template/components/base";
+import {
+  getBooleanParamFromTemplateRow,
+  getStringParamFromTemplateRow,
+} from "src/app/shared/utils";
+
+interface IExampleComponentParams {
+  /** TEMPLATE_PARAMETER: "variant". Default "red_square" */
+  variant: "red_square" | "yellow_circle";
+  /** TEMPLATE_PARAMETER: "my_boolean_param". A paramater that takes a boolean value. Default true */
+  myBooleanParam: boolean;
+  /** TEMPLATE_PARAMETER: "my_string_param". A string to be displayed */
+  myStringParam: string;
+}
+
+@Component({
+  selector: "plh-example",
+  templateUrl: "./example.component.html",
+  styleUrls: ["./example.component.scss"],
+})
+export class TmplExampleComponent extends TemplateBaseComponent implements OnInit {
```

Comment:
nit(blocking): I think we should probably drop the `Tmpl` prefix convention so that new component collections aren't as likely to confilct with each other. I'd say better to prefix with the namespace I've moved the component under, i.e. `DemoBasicComponent`